### PR TITLE
Feature/webserver revamp

### LINKED ISSUE #288: failure to compile on linux
i am unable to build performous (current git) on arch linux:

```
[ 14%] Building CXX object game/CMakeFiles/performous.dir/webserver.cc.o
/home/jose/tarballs/aur4/performous-git/src/performous-git/game/webserver.cc:351:1: error: redefinition of ‘WebServer::WebServer(Songs&)’
 WebServer::WebServer(Songs& songs)
 ^~~~~~~~~
In file included from /home/jose/tarballs/aur4/performous-git/src/performous-git/game/webserver.cc:1:0:
/home/jose/tarballs/aur4/performous-git/src/performous-git/game/webserver.hh:63:2: note: ‘WebServer::WebServer(Songs&)’ previously defined here
  WebServer(Songs&) {}
  ^~~~~~~~~
/home/jose/tarballs/aur4/performous-git/src/performous-git/game/webserver.cc:353:23: error: definition of implicitly-declared ‘WebServer::~WebServer()’
 WebServer::~WebServer(){}
                       ^
make[2]: *** [game/CMakeFiles/performous.dir/build.make:1575: game/CMakeFiles/performous.dir/webserver.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:159: game/CMakeFiles/performous.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

any idea?

### LINKED ISSUE #306: Upgrade to latest cppnetlib Or use boost
We should upgrade from cppnetlib 0.11.2 to the latest stable (0.12.0) 
Docs for 0.12.0 are to find here: http://cpp-netlib.org/0.12.0/index.html 

Or while at it completely remove the cppnetlib library since it's not supported anymore in ubuntu > 17.10
and drop back to use boost::asio (http://www.boost.org/doc/libs/1_65_1/doc/html/boost_asio/examples/cpp11_examples.html )

Personal favorite goes to drop cppnetlib and use `boost::asio` since we're already using lots of boost and it's still in the repo's.

Related issue: #288 

Vote :heart: (using `boost::asio`) vs :+1: (upgrading cppnetlib)

### LINKED ISSUE #369: Ok seriously... we need to drop cpp-netlib.
I opened another issue because I think we need to retake this discussion and the best way to do it is starting fresh.

cpp-netlib is a fucking mess; 0.11.2 won't compile at all on boost >= 1.66; and the tests won't build on 1.65 either due to some issue with the Boost Testing framework. I can get it to build if I don't run tests but that's hardly ideal.
cpp-netlib 0.12 changed to standalone Asio instead of Boost.Asio, then 0.13 (currently in rc1 since god knows when) changed back to Boost.Asio.
 
The cpp-netlib 0.13 source tarball refuses to build (cmake errors) because they included dependencies like Google Test framework but didn't put them in the tarball (you're supposed to git clone and download submodules)... not to mention the project is as good as abandoned.

I was considering taking on upgrading performous to 0.13 but frankly it'd be backing ourselves deeper into a corner. We need alternatives. Boost.Asio is too low-level, sure. Perhaps Boost.Beast? It's still a lower-level than cpp-netlib but it's build on top of Boost.Asio and is under active development.

Other options? Would maybe something like mongoose work?

Suggestions/opinions please?

### PR DESCRIPTION
**Pull request by Baklap4**
### What does this PR do?

* Replaces `cppnetlib` with [`libcpprest`](https://github.com/Microsoft/cpprestsdk/tree/v2.10.2)
* Adds a request handler to care about incoming requests.
* Fixes some JS ajax calls due to the content-type send by it as default is `application/x-www-form-urlencoded` while i was sending json. The server didn't like that so i changed it to `application/json`

### Closes Issue(s)

* Closes #369 
* Closes #306 
* Closes #288 

### Motivation

Cppnetlib was/is dropped from the repo's and it was outdated (we were using 0.11.2 while 0.13 was already out). Besides this people had trouble building with the webserver which should be solved by this.

### Additional Notes

For now i've just copied over the original functionality and made sure everything still works as it did.
MXE has no file for cpprestsdk. I'll create it somewhere this week i hope so we can build for windows aswell again.

### CODE REVIEW

**Comment by Baklap4** [comment]
@Lord-Kamina the latest and greatest: https://github.com/Microsoft/cpprestsdk/tree/v2.10.2 as mentioned in my pr (libcpprest link)

File: game/webserver.cc [hunk 1]
```diff
@@ -83,291 +47,9 @@ WebServer::WebServer(Songs& songs)
...
-			return m_response;
-
-		}
-		catch(std::exception e) {
-			return http_server::response::stock_reply(http_server::response::not_found, "notfound");
-		}
-	}
-}
-
-http_server::response WebServer::POSTresponse(const http_server::request &request) {
```
[the remark this answers is not in the document]
Comment (Baklap4) [reply game/webserver.cc]: I've removed the comment :+1:

**Comment by Lord-Kamina** [comment]
BTW I hate you so much  now. Yesterday I was all like "Shit I need to get my shit together so I can finish with RESTinio before he does it on cpprestsdk" then I went to the cinema, came back and this PR was up 😂

**Comment by Baklap4** [comment]
Lmao @Lord-Kamina  :joy: It took me 1 day to completely rewrite it, mostly due to cmake not linking / finding shit. Then another day cleaning stuff, and currently it has all the features we had already including some extra checking for valid requests :+1: 

Since our CI is running ubuntu 14.04 i had to add some build steps for building the package myself. Personally i found that kind of meh but it works

**Comment by Lord-Kamina** [comment]
I'll still probably make a PR changing to rapidjson in the next couple days though.

**Pull request merged**